packages/access/graphql/schema.js

```diff
@@ -25,14 +25,18 @@ type mutations {
   """
   Claim a granted membership with a voucher code
   """
-  claimAccess(voucherCode: String!): AccessGrant!
+  claimAccess(
+    voucherCode: String!
+    payload: AccessGrantPayloadInput
```

Comment:
```suggestion
    payload: JSON
```
I would just accept arbitrary json. This would allow us to start tracking new stuff without a backend change. E.g. add referrers or other stuff as we see fit. And since the whole object is optional there is no guarantee anyhow.